tests/test_sealed_box.py

```diff
@@ -111,20 +115,23 @@ def test_wrong_types():
         priv.public_key.encode(),
     )
     with pytest.raises(TypeError):
-        SealedBox(priv, priv.public_key)
+        # Type safety: we want to check this error is detected at runtime.
+        SealedBox(priv, priv.public_key)  # type: ignore[call-arg]
 
 
 @pytest.mark.parametrize(
     ("_privalice", "pubalice", "_plaintext", "encrypted"), sealbox_vectors()
 )
 def test_sealed_box_public_key_cannot_decrypt(
-    _privalice, pubalice, _plaintext, encrypted
+    _privalice: bytes, pubalice: bytes, _plaintext: bytes, encrypted: bytes
 ):
-    pubalice = PublicKey(pubalice, encoder=HexEncoder)
+    pubalice_decoded = PublicKey(pubalice, encoder=HexEncoder)
 
-    box = SealedBox(pubalice)
+    box = SealedBox(pubalice_decoded)
     with pytest.raises(TypeError):
-        box.decrypt(
+        # Type safety: mypy spots that you can't decrypt with a public key, but we
+        # want to detect this at runtime too.
```

Comment:
I was pleased to see this caught by mypy---I'd forgotten that we did this in #706!